src/components/HistoryTables/MessageHistory/Detail/index.tsx

```diff
@@ -50,26 +42,33 @@ export default function MessageDetail(props: MessageDetailProps) {
   const { isProd } = useEnvironment()
   const logger = useLogger()
 
-  const { data: gasQuery, error: gasStateError } = useGasCostQuery({
-    variables: { cid },
-    pollInterval: 10000
-  })
-  const { data: msgRcptQuery, error: msgRcptError } = useMessageReceiptQuery({
-    variables: { cid },
-    pollInterval: 10000
-  })
+  const { data: stateReplayQuery, error: stateReplayError } =
+    useStateReplayQuery({
+      variables: { cid: props.cid },
+      pollInterval: 30000,
+      fetchPolicy: 'cache-first',
+      // give the message time to execute on-chain before fetching
+      skip: !props.cid || props.confirmations < 2
```

Comment:
This `!props.cid` check could _probably_ be skipped because the cid prop is required and ts will probably catch this error in any app. But i think it's fine to keep just to limit any mistakes and hit the node with invalid requests to an expensive endpoint